src/native/fs/fs_napi.cpp

```diff
@@ -1736,21 +1712,50 @@ struct UnlinkFileAt : public FSWrapWorker<FileWrap>
 {
     std::string _filepath;
     int _delete_fd;
+    int64_t _verify_inode;
+    int _flags;
     UnlinkFileAt(const Napi::CallbackInfo& info)
         : FSWrapWorker<FileWrap>(info)
         , _delete_fd(-1)
+        , _verify_inode(-1)
+        , _flags(0)
     {
         _filepath = info[1].As<Napi::String>();
         if (info.Length() > 2 && !info[2].IsUndefined()) {
             _delete_fd = info[2].As<Napi::Number>();
         }
-        Begin(XSTR() << "UnlinkFileAt" << DVAL(_wrap->_path.c_str()) << DVAL(_wrap->_fd) << DVAL(_filepath) << DVAL(_delete_fd));
+        if (info.Length() > 3 && !info[3].IsUndefined()) {
+            _verify_inode = info[3].As<Napi::Number>().Int64Value();
+        }
+        if (info.Length() > 4 && !info[4].IsUndefined()) {
+            _flags = info[4].As<Napi::Number>();
+        }
+        Begin(XSTR() << "UnlinkFileAt" << DVAL(_wrap->_path.c_str()) << DVAL(_wrap->_fd) << DVAL(_filepath) << DVAL(_delete_fd) << DVAL(_verify_inode));
     }
     virtual void Work()
     {
         int fd = _wrap->_fd;
+        std::string path = _wrap->_path;
         CHECK_WRAP_FD(fd);
-        SYSCALL_OR_RETURN(dlsym_gpfs_unlinkat(fd, _filepath.c_str(), _delete_fd));
+        if(use_gpfs_lib()) {
+            SYSCALL_OR_RETURN(dlsym_gpfs_unlinkat(fd, _filepath.c_str(), _delete_fd));
+        } else {
+            DBG1("Unlinkat " << DVAL(fd) << DVAL(path.c_str()) << DVAL(_filepath.c_str()) << DVAL(_flags) << DVAL(_verify_inode));
+            if (_verify_inode > 0) {
+                struct stat _stat_res;
+                std::string full_path = path + "/" + _filepath;
+                DBG1("Unlinkat " << DVAL(full_path));
+                SYSCALL_OR_RETURN(stat(full_path.c_str(), &_stat_res));
+                int64_t stat_actual_ino = _stat_res.st_ino;
+                DBG1("Unlinkat " << DVAL(stat_actual_ino) << DVAL(_verify_inode));
+                if (stat_actual_ino != _verify_inode) {
```

Comment:
we should probably add comment about this being best effort. file could still change between stat and unlink